src/Rings/mpoly-localizations.jl

```diff
@@ -92,9 +92,28 @@ Base.iterate(U::MPolyPowersOfElement, i::Int) = (i < length(U.a) ? (U.a[i+1], i+
 is_trivial(U::MPolyPowersOfElement) = (U == units_of(ambient_ring(U)))
 
 ### printing
+function Base.show(io::IO, ::MIME"text/plain", S::MPolyPowersOfElement)
+  io = pretty(io)
+  println(io, "Multiplicative subset")
+  print(io, Indent())
+  println(io, "of ", Lowercase(), ambient_ring(S))
+  print(io, "given by the powers of ")
+  print(io, "[")
+  join(io, denominators(S), ", ")
+  print(io, "]")
```

Comment:
This can get really lengthy. May take the same route as for ideals: at powers/products of n elements